### PR TITLE
Add mixed16 tests to point_to_point communications

### PR DESCRIPTION
**Pull request by belldandyxtq**
This commit adds mixed16 tests to all the p2p communications

Thank you for creating a pull request!

Please double-check the following.

- Read [our contribution guide](https://docs.chainer.org/en/stable/contribution.html).
  - Does your code conform to our coding guidelines?
  - Did you write sufficient test code?
  - Did you write sufficient documentation?
- Also, take a look at [our compatibility policy](https://docs.chainer.org/en/stable/compatibility.html).

### CODE REVIEW

**Comment by belldandyxtq** [comment]
pfnci, test this please

**Comment by belldandyxtq** [comment]
pfnci, test this please

File: tests/chainermn_tests/functions_tests/test_point_to_point_communication.py [hunk 1]
```diff
@@ -44,8 +46,9 @@ def create_communicator(gpu, param):
 
 
 def create_x(gpu, param, communicator):
+    io_dtype = chainer.get_dtype(dtype=param.dtype, map_mixed16=numpy.float16)
```
Comment (kuenishi) [inline tests/chainermn_tests/functions_tests/test_point_to_point_communication.py]: What is intended by explicitly setting `map_mixed16` here?

File: tests/chainermn_tests/functions_tests/test_point_to_point_communication.py [hunk 2]
```diff
@@ -239,21 +242,22 @@ def test_non_variable_send(param):
 
     rank_send = (communicator.rank + 1) % communicator.size
     rank_recv = (communicator.rank - 1) % communicator.size
+    io_dtype = chainer.get_dtype(dtype=param.dtype, map_mixed16=numpy.float16)
```
Comment (kuenishi) [inline tests/chainermn_tests/functions_tests/test_point_to_point_communication.py]: ditto

File: tests/chainermn_tests/functions_tests/test_point_to_point_communication.py (hunk 1 above)
Comment (belldandyxtq) [reply tests/chainermn_tests/functions_tests/test_point_to_point_communication.py]: The same as the other PR

**Comment by kuenishi** [comment]
Jenkins, test this please.

**Pull request merged**